### PR TITLE
TST: add test for get_loc on tz-aware DatetimeIndex

### LINKED ISSUE #32140: Pandas 1.0.1 fails get_loc on timeindex (0.25.3 works)
```python
findtime = pd.Timestamp('2019-12-12 10:19:25', tz='US/Eastern')
start = pd.Timestamp('2019-12-12 0:0:0', tz='US/Eastern')
end = pd.Timestamp('2019-12-13 0:0:0', tz='US/Eastern')
testindex = pd.date_range(start, end, freq='5s')
testindex.get_loc(findtime, method='nearest')

```
#### Problem description
With pandas 1.0.1, python 3.8.1

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python38\lib\site-packages\pandas\core\indexes\datetimes.py", line 699, in get_loc
    return Index.get_loc(self, key, method, tolerance)
  File "C:\Program Files\Python38\lib\site-packages\pandas\core\indexes\base.py", line 2649, in get_loc
    indexer = self.get_indexer([key], method=method, tolerance=tolerance)
  File "C:\Program Files\Python38\lib\site-packages\pandas\core\indexes\base.py", line 2740, in get_indexer
    indexer = self._get_nearest_indexer(target, limit, tolerance)
  File "C:\Program Files\Python38\lib\site-packages\pandas\core\indexes\base.py", line 2821, in _get_nearest_indexer
    left_distances = abs(self.values[left_indexer] - target)
numpy.core._exceptions.UFuncTypeError: ufunc 'subtract' cannot use operands with types dtype('<M8[ns]') and dtype('O')
```

With 0.25.3, this returns index location 7433 as expected.

I've tested this on Windows 10 1909, and Ubuntu WSL, and get the same error on both.

#### Output of ``pd.show_versions()``

```
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_Canada.1252

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 41.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None
```

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #32140
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_indexing.py [hunk 1]
```diff
@@ -452,6 +452,17 @@ def test_get_loc_reasonable_key_error(self):
...
+        # https://github.com/pandas-dev/pandas/issues/32140
+        dti = pd.date_range(
+            pd.Timestamp("2019-12-12 00:00:00", tz="US/Eastern"),
+            pd.Timestamp("2019-12-13 00:00:00", tz="US/Eastern"),
+            freq="5s",
+        )
+        key = pd.Timestamp("2019-12-12 10:19:25", tz="US/Eastern")
+        result = dti.get_loc(key, method="nearest")
+        assert result == 7433
+
```
Comment (jbrockmendel) [inline pandas/tests/indexes/datetimes/test_indexing.py]: nitpick (not a blocker), can you put this a few tests up adjacent to the other tests that use the `method` kwarg

Comment (simonjayhawkins) [reply pandas/tests/indexes/datetimes/test_indexing.py]: sure. done.

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/tests/indexes/datetimes/test_indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/datetimes/test_indexing.py]: thanks

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Comment by jreback** [comment]
@simonjayhawkins may need a manual backport

**Pull request merged**